src/System.Reflection.TypeExtensions/tests/EventInfoTests.cs

```diff
@@ -0,0 +1,184 @@
+        {
+            // Add and make sure we bound the event.
+            eventInfo.AddEventHandler(target, handler);
+            (target ?? new EI_Class()).InvokeAllEvents();
+            Assert.Equal(expectedStaticVariable, EI_Class.StaticVariable);
+            EI_Class.StaticVariable = 0; // Reset
+
+            // Remove and make sure we unbound the event.
+            eventInfo.RemoveEventHandler(target, handler);
+            (target ?? new EI_Class()).InvokeAllEvents();
+            Assert.Equal(0, EI_Class.StaticVariable);
+        }
+
+        public static IEnumerable<object[]> AddEventHandler_Invalid_TestData()
+        {
+            // Null target for instance method
+            EI_Class tc1 = new EI_Class();
+            yield return new object[] { typeof(EI_Class).GetEvent("PublicEvent"), null, new VoidDelegate(tc1.ProtectedInternalVoidMethod), typeof(TargetException) };
+
+            // Event not declared on target
+            yield return new object[] { typeof(EI_Class).GetEvent("PublicEvent"), new DummyClass(), new VoidDelegate(tc1.ProtectedInternalVoidMethod), typeof(TargetException) };
+
+            // Event does not have a public add accessor
+            yield return new object[] { typeof(EI_Class).GetEvent("PrivateEvent", BindingFlags.NonPublic | BindingFlags.Instance), new DummyClass(), new VoidDelegate(tc1.ProtectedInternalVoidMethod), typeof(InvalidOperationException) };
+        }
+        
+        [Theory]
+        [MemberData(nameof(AddEventHandler_Invalid_TestData))]
+        public void AddEventHandler_Invalid(EventInfo eventInfo, object target, Delegate handler, Type exceptionType)
+        {
+            Assert.Throws(exceptionType, () => eventInfo.AddEventHandler(target, handler));
+            Assert.Throws(exceptionType, () => eventInfo.RemoveEventHandler(target, handler));
+        }
+
+        [Theory]
+        [InlineData("PublicEvent")]
+        [InlineData("Public_Event")]
+        [InlineData("Private_Event")]
+        [InlineData("PrivateEvent")]
+        [InlineData("ProtectedEvent")]
+        [InlineData("ProtectedInternalEvent")]
+        [InlineData("ProtectedInternalEvent")]
+        public void Attributes_IsSpecialName(string name)
+        {
+            EventInfo eventInfo = Helpers.GetEvent(typeof(EI_Class), name);
+            Assert.Equal(EventAttributes.None, eventInfo.Attributes);
+            Assert.False(eventInfo.IsSpecialName);
+        }
+
+        [Theory]
+        [InlineData("PublicEvent", typeof(VoidDelegate))]
+        [InlineData("PrivateEvent", typeof(VoidDelegate))]
+        [InlineData("ProtectedEvent", typeof(VoidDelegate))]
+        [InlineData("ProtectedInternalEvent", typeof(VoidDelegate))]
+        public void EventHandlerType(string name, Type expected)
+        {
+            EventInfo eventInfo = Helpers.GetEvent(typeof(EI_Class), name);
+            Assert.Equal(expected, eventInfo.EventHandlerType);
```

Comment:
I seem to remember doing this, and getting stumped because Assert.IsType for a type parameter returns RuntimeType. This is because IsType goes one level deeper - the type of a type is RuntimeType